### PR TITLE
refactor: extract re-usable code from poetry fix

### PR DESCRIPTION
**Pull request by lili2311**
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules

#### What does this PR do?
In preparation for introducing a different sequential fix strategy
refactor the existing code into re-usable blocks needed to add sequential
fix. 
Added some extra optional fields that can be sent in changes so we have all the data we need in case we want to output is as --json or consume it another way

#### Where should the reviewer start?
packages/snyk-fix/src/plugins/python/handlers/poetry/update-dependencies/index.ts

### CODE REVIEW

File: packages/snyk-fix/src/plugins/python/handlers/poetry/update-dependencies/index.ts [hunk 1]
```diff
@@ -128,56 +111,82 @@ function throwPoetryError(stderr: string, command?: string) {
...
+      changes,
+    });
   } catch (error) {
     debug(
       `Failed to fix ${entity.scanResult.identity.targetFile}.\nERROR: ${error}`,
     );
     handlerResult.failed.push({
       original: entity,
       error,
-      tip: poetryCommand ? `Try running \`${poetryCommand}\`` : undefined,
```
Comment (aarlaud) [inline packages/snyk-fix/src/plugins/python/handlers/poetry/update-dependencies/index.ts:179]: we just nuke that tip thing? I'm not seeing it moved elsewhere

Comment (aarlaud) [reply packages/snyk-fix/src/plugins/python/handlers/poetry/update-dependencies/index.ts:179]: nevermind ;)

**Review by aarlaud — APPROVED** [review]

File: packages/snyk-fix/src/plugins/python/handlers/poetry/update-dependencies/index.ts (hunk 1 above)
Comment (lili2311) [reply packages/snyk-fix/src/plugins/python/handlers/poetry/update-dependencies/index.ts:179]: good catch, we do want to keep it. Let me see what `tip` makes sense here as we lose the context of the command we ran.

Comment (lili2311) [reply packages/snyk-fix/src/plugins/python/handlers/poetry/update-dependencies/index.ts:179]: added the tip via the Error thrown

**Pull request merged**